### PR TITLE
feat(util):  Add support for Win32 URIs

### PR DESCRIPTION
**Pull request by GordonSmith**
Signed-off-by: Gordon Smith <GordonJSmith@gmail.com>

## Checklist:
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit message includes a "fixes" reference if appropriate.
  - [x] The commit is signed.
- [x] The change has been fully tested:
  - [ ] I have viewed all related gallery items
  - [ ] I have viewed all related dermatology items
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised new issues to address them separately

### CODE REVIEW

File: packages/util/src/url.ts [hunk 1]
```diff
@@ -1,27 +1,46 @@
...
+                resultParts.pop();
+                continue;
+            }
+            resultParts.push(part);
         }
-        resultParts.push(part);
+        return resultParts.join(separator);
+    }
+
+    function dirname(path: string) {
```
Comment (jeclrsg) [inline packages/util/src/url.ts:31]: Reminder to check the naming convention & functionality of this vs Node

File: tests/test-util/src/url.spec.ts [hunk 2]
```diff
@@ -24,4 +33,39 @@ describe("url", function () {
...
+
+        expect(win32.join(".", "bbb", "ccc")).to.equal("bbb\\ccc");
+        expect(win32.join("\\.", "bbb", "ccc")).to.equal("\\bbb\\ccc");
+        expect(win32.join(".\\", "bbb", "ccc")).to.equal("bbb\\ccc");
+        expect(win32.join("\\.\\", "bbb", "ccc")).to.equal("\\bbb\\ccc");
+    });
+
+    it("win32 dirname", function () {
+        expect(win32.dirname("\\aaa\\bbb\\ccc")).to.equal("\\aaa\\bbb");
+        expect(win32.dirname("\\aaa\\bbb\\ccc\\")).to.equal("\\aaa\\bbb");
```
Comment (jeclrsg) [inline tests/test-util/src/url.spec.ts:68]: And then, dependent upon the previous comment, should this equal "\\aaa\\bbb\\ccc\\" or possibly without the final dir separator?

**Review by jeclrsg — CHANGES_REQUESTED** [review]
@GordonSmith A couple comments about dirname

File: packages/util/src/url.ts (hunk 1 above)
Comment (GordonSmith) [reply packages/util/src/url.ts:32]: I added some tests for this and indeed there are some discrepancies...  

I think the solution is to take https://github.com/browserify/path-browserify and update it with the latest LTS NodeJS version?

**Pull request closed without merging**